binlog_streamer.go

```diff
@@ -241,26 +251,35 @@ func (s *BinlogStreamer) FlushAndStop() {
 	// passed the initial target position.
 	err := WithRetries(100, 600*time.Millisecond, s.logger, "read current binlog position", func() error {
 		var err error
-		s.targetBinlogPosition, err = ShowMasterStatusBinlogPosition(s.DB)
+		s.stopAtBinlogPosition, err = ShowMasterStatusBinlogPosition(s.DB)
 		return err
 	})
 
 	if err != nil {
 		s.ErrorHandler.Fatal("binlog_streamer", err)
 	}
-	s.logger.WithField("target_position", s.targetBinlogPosition).Info("current stop binlog position was recorded")
+	s.logger.WithField("target_position", s.stopAtBinlogPosition).Info("current stop binlog position was recorded")
 
 	s.stopRequested = true
 }
 
-func (s *BinlogStreamer) updateLastStreamedPosAndTime(ev *replication.BinlogEvent) {
+func (s *BinlogStreamer) updateLastStreamedPosAndTime(ev *replication.BinlogEvent, isResumablePosition bool) {
 	if ev.Header.LogPos == 0 || ev.Header.Timestamp == 0 {
 		// This shouldn't happen, as the cases where it does happen are excluded.
 		// However, I've not seen all the cases yet.
 		s.logger.Panicf("logpos: %d %d %T", ev.Header.LogPos, ev.Header.Timestamp, ev.Event)
 	}
 
-	s.lastStreamedBinlogPosition.Pos = ev.Header.LogPos
+	pos := mysql.Position{
+		Name: s.currentBinlogFilename,
+		Pos:  ev.Header.LogPos,
+	}
+
+	s.lastStreamedBinlogPosition = pos
+	if isResumablePosition {
+		s.lastResumableBinlogPosition = pos
+	}
+
```

Comment:
I think the cleanup here is nice 👍 Consolidating all of this in one place definitely makes is more clear.